### PR TITLE
[SAI-PTF] Refactor code for more generic service and dut data modeling  

### PR DESCRIPTION
**Pull request by ms-junyi**
Signed-off-by: ms-junyi <t-junyixiao@microsoft.com>

### CODE REVIEW

File: test/sai_test/config/route_configer.py [hunk 1]
```diff
@@ -29,15 +29,15 @@ def t0_route_config_helper(test_obj, is_create_route=True, is_create_route_for_l
         route_configer.create_default_route()
 
     if is_create_route_for_lag:
-        route_configer.create_route_and_neighbor_entry_for_port(ip_addr=test_obj.lag1_ip,
-            mac_addr=test_obj.lag1_nb_mac,
-            port_id=test_obj.lag1.lag_id,
-            virtual_router_id=test_obj.default_vrf)
+        route_configer.create_route_and_neighbor_entry_for_port(ip_addr=test_obj.lag1_neighbor.ipv4,
```
Comment (Gfrom2016) [inline test/sai_test/config/route_configer.py:32]: please rebase the newest master branch and make change based on that, route and neighbor config for lag has changed.

File: test/sai_test/constant.py [hunk 2]
```diff
@@ -40,7 +40,11 @@
...
+FDB_T0_NUM = '00'
 
-SERVER_IP_PREFIX = '192.168.{}.{}'
-T0_IP_PREFIX = '10.0.{}.{}'
-T1_IP_PREFIX = '10.1.{}.{}'
+SERVER_IPV4_PREFIX = '192.168.{}.{}'
+T0_IPV4_PREFIX = '10.0.{}.{}'
+T1_IPV4_PREFIX = '10.1.{}.{}'
+
+ROUTER_MAC = '00:77:66:55:44:00'
```
Comment (Gfrom2016) [inline test/sai_test/constant.py:50]: already exist in this file at line 30

File: test/sai_test/sai_test_base.py [hunk 3]
```diff
@@ -214,6 +217,18 @@ class T0TestBase(ThriftInterfaceDataPlane):
...
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dut = Dut()
+        self.num_device_each_group = 99
+        self.server_group0, self.server_group1, self.server_group2, self.server_group3 = [], [], [], []
+        self.t0_group1, self.t0_group2, self.t0_group3 = [], [], []
+        self.t1_group1, self.t1_group2, self.t1_group3 = [], [], []
+        self.create_device()
+        self.lag1_neighbor = self.t1_group1[0]
```
Comment (Gfrom2016) [inline test/sai_test/sai_test_base.py:228]: Is it better to put the assignment of these two variables in lag_config.py after create lag?

File: test/sai_test/sai_test_base.py [hunk 4]
```diff
@@ -353,3 +332,7 @@ def tearDown(self):
         t0_vlan_tear_down_helper(self)
         t0_port_tear_down_helper(self)
         super().tearDown()
+
+if __name__ == '__main__':
+    a = T0TestBase()
+    a.setUp()
```
Comment (Gfrom2016) [inline test/sai_test/sai_test_base.py:338]: newline at the end of file please

File: test/sai_test/sai_lag_test.py [hunk 5]
```diff
@@ -34,29 +34,33 @@ def setUp(self):
         T0TestBase.setUp(self)
 
     def load_balance_on_src_ip(self):
-        ip_dst = self.lag1_route_dst
+        sai_thrift_create_router_interface(self.client,
```
Comment (Gfrom2016) [inline test/sai_test/sai_lag_test.py:37]: No need to recreate this interface for port1 cause it has been create in route_configer.py

**Review by richardyu-ms — APPROVED** [review]

**Pull request merged**